### PR TITLE
Fix arc-analyze-prologue test and add support of frames unwinder for HS5x

### PR DESCRIPTION
**Pull request by kolerov**
There is a series of patches:

1. `arc: Don't build arc-analyze-prologue.S with -g`
   Fix `arc-analyze-prologue` test itself. It's already pushed to upstream.
2. `arc64: opcodes: Fix instruction class for push/pop`
   Fix push/pop opcodes for HS6x. It will be useful for future frame unwinder implementation for HS6x.
3. `arc64: gdb: Fix an order of registers for ENTER`
   Fix ENTER instruction handler in the frame unwinder for HS5x.
4. `arc64: gdb: Add support of prologue tests for HS5x`

The latest GDB with those patches passes `arc-analyze-prologue` test for ARCv2 and HS5x targets.

### CODE REVIEW

File: gdb/arc64-tdep.c
Comment (shahab-vahedi) [inline gdb/arc64-tdep.c:1]: nitpick: The word "instructions" in the first line of commit message should change to "registers".

File: gdb/arc64-tdep.h [hunk 1]
```diff
@@ -41,10 +41,10 @@ enum arc_regnum
   ARC_R7_REGNUM = 7,
   ARC_R9_REGNUM = 9,
   ARC_R13_REGNUM = 13,
```
Comment (shahab-vahedi) [inline gdb/arc64-tdep.h]: Maybe `ARC_R13_REGNUM = 13` can be removed?

File: gdb/testsuite/gdb.arch/arc-analyze-prologue.S
Comment (shahab-vahedi) [inline gdb/testsuite/gdb.arch/arc-analyze-prologue.S:1]: I think `arcv3:32 (HS5x)` can have `lp` support. Please check the PRM.

**Review by shahab-vahedi — COMMENTED** [review]
The changes are mostly OK, but please reconfirm the situation for `lp` support in HS5x.

File: gdb/arc64-tdep.h (hunk 1 above)
Comment (kolerov) [reply gdb/arc64-tdep.h]: Yes, you are right. It may be safely removed.

File: gdb/testsuite/gdb.arch/arc-analyze-prologue.S
Comment (kolerov) [reply gdb/testsuite/gdb.arch/arc-analyze-prologue.S:1]: Indeed! I've found that loops are supported by HS5x but not by HS6x. However, they are not implemented yet in GNU tools for HS5x. So, I marked them as `xfail`.

File: gdb/testsuite/gdb.arch/arc-analyze-prologue.S [hunk 2]
```diff
@@ -155,14 +155,18 @@ noncallee_saved_regs_gp_push:
 	add	sp,sp,8
 
 ; LP_COUNT is treated like a normal register.
+; HS6x does not have loop instructions and registers but HS5x does. However,
+; it's nop supported by GNU tools yet.
```
Comment (shahab-vahedi) [inline gdb/testsuite/gdb.arch/arc-analyze-prologue.S]: typo: nop -> not

File: gdb/testsuite/gdb.arch/arc-analyze-prologue.S [hunk 3]
```diff
@@ -415,7 +423,10 @@ predicated_insn:
 	j	[blink]
 
 ; Loops should halt prologue analysis.
+; HS6x does not have loop instructions and registers but HS5x does. However,
+; it's nop supported by GNU tools yet.
```
Comment (shahab-vahedi) [inline gdb/testsuite/gdb.arch/arc-analyze-prologue.S]: typo: nop -> not

File: gdb/testsuite/gdb.arch/arc-analyze-prologue.exp [hunk 4]
```diff
@@ -109,7 +130,15 @@ prologue_test "noncallee_saved_regs_r12_st" { {r12 0} {r13 4} }
 prologue_test "noncallee_saved_regs_r12_push" { {r12 0} {r13 4} }
 prologue_test "noncallee_saved_regs_r2_push" { {r2 0} {r13 4} }
 prologue_test "noncallee_saved_regs_gp_push" { {r25 4} {gp 0} }
-prologue_test "noncallee_saved_regs_lp_count" { {r25 4} {lp_count 0} }
+# HS6x does not have loop instructions and registers but HS5x does. However,
+# it's nop supported by GNU tools yet.
```
Comment (shahab-vahedi) [inline gdb/testsuite/gdb.arch/arc-analyze-prologue.exp]: typo: nop -> not

File: gdb/testsuite/gdb.arch/arc-analyze-prologue.exp [hunk 5]
```diff
@@ -154,7 +197,15 @@ prologue_test "cond_branch_in_prologue" { {r13 4} }
 prologue_test "jump_in_prologue" { {r13 0} }
 prologue_test "cond_jump_in_prologue" { {r13 4} }
 prologue_test "predicated_insn" { {r13 8} {r15 0} }
-prologue_test "loop_in_prologue" { {r25 4} {lp_count 0} }
+# HS6x does not have loop instructions and registers but HS5x does. However,
+# it's nop supported by GNU tools yet.
```
Comment (shahab-vahedi) [inline gdb/testsuite/gdb.arch/arc-analyze-prologue.exp]: typo: nop -> not

**Review by shahab-vahedi — APPROVED** [review]

**Pull request merged**